### PR TITLE
Fixed the issue for Linux Build

### PR DESCRIPTION
**Pull request by jianyunt**
Fixed the issue for Linux Build
Sync'ed up  with PowerShellCore OneGet
Added PowerShellGet as submodule
On PowerShellCore, OneGet should not search ProviderAssemblies folder for loading provider assemblies

### CODE REVIEW

**Comment by jianyunt** [comment]
@bmanikm @KrishnaV-MSFT  , can you perform CR? thx

File: src/Microsoft.PackageManagement/Implementation/PackageManagementService.cs [hunk 1]
```diff
@@ -1038,28 +1037,28 @@ private IMetaProvider GetMetaProviderObject(IHostApi request)
...
-            var providerAssemblies = (_initialized ? Enumerable.Empty<string>() : _defaultProviders)
-                .Concat(GetProvidersFromRegistry(Registry.LocalMachine, "SOFTWARE\\MICROSOFT\\PACKAGEMANAGEMENT"))
-                .Concat(GetProvidersFromRegistry(Registry.CurrentUser, "SOFTWARE\\MICROSOFT\\PACKAGEMANAGEMENT"));
+            var providerAssemblies = Enumerable.Empty<string>();
+#if !UNIX
+            //On PowerShell FullCLR , we will need to search providerassemblies folder
+            var pshome = RunPowerShellCommand(request, "$PSHome");
+            if (!string.IsNullOrWhiteSpace(pshome) && pshome.EndsWith(@"\WindowsPowerShell\v1.0", StringComparison.OrdinalIgnoreCase))
+            {
+                request.Debug("Current running environment: PowerShell FullCRL.");
```
Comment (bmanikm) [inline src/Microsoft.PackageManagement/Implementation/PackageManagementService.cs]: >PowerShell FullCRL [](start = 60, length = 18)

Consider using 'Windows PowerShell'

File: src/Microsoft.PackageManagement/Implementation/PackageManagementService.cs [hunk 2]
```diff
@@ -1116,6 +1115,33 @@ private IMetaProvider GetMetaProviderObject(IHostApi request)
...
+                using (PowerShell powershell = PowerShell.Create())
+                {
+                    try
+                    {
+                        rs.Open();
+                        powershell.Runspace = rs;
+                        powershell.AddScript(commandName);                                    
+                        var f = powershell.Invoke().FirstOrDefault();
+
+                        return f.ToString();
```
Comment (bmanikm) [inline src/Microsoft.PackageManagement/Implementation/PackageManagementService.cs]: Handle the error cases

File: Test/run-tests.ps1 [hunk 3]
```diff
@@ -267,8 +271,27 @@ if ($testframework -eq "coreclr")
...
+    Write-Verbose ("PowerShellGet Folder '{0}'" -f $PSGetPath)
+
+    if(-not (Test-Path -Path $PSGetPath))
+    {
+        New-Item -Path $PSGetPath -ItemType Directory -Force -Verbose
+    }
+
+
+    # Copying files to Packagemanagement and PowerShellGet folders
+    Copy-Item "$PowerShellGetPath\*" $PSGetPath -force -verbose
```
Comment (bmanikm) [inline Test/run-tests.ps1]: >Copy-Item "$PowerShellGetPath\*" $PSGetPath -force -verbose [](start = 4, length = 59)

Consider adding -Recursive

**Comment by bmanikm** [comment]
:shipit:

File: src/Microsoft.PackageManagement/Implementation/PackageManagementService.cs (hunk 1 above)
Comment (jianyunt) [inline src/Microsoft.PackageManagement/Implementation/PackageManagementService.cs]: fixed

---
In reply to: [82307593](https://github.com/OneGet/oneget/pull/218#discussion_r82307593) [](ancestors = 82307593)

File: Test/run-tests.ps1 (hunk 3 above)
Comment (jianyunt) [inline Test/run-tests.ps1]: added

---
In reply to: [82308509](https://github.com/OneGet/oneget/pull/218#discussion_r82308509) [](ancestors = 82308509)

File: src/Microsoft.PackageManagement/Implementation/PackageManagementService.cs (hunk 2 above)
Comment (jianyunt) [inline src/Microsoft.PackageManagement/Implementation/PackageManagementService.cs]: added

---
In reply to: [82308037](https://github.com/OneGet/oneget/pull/218#discussion_r82308037) [](ancestors = 82308037)

**Pull request merged**